<?xml version="1.0"?>

<launch>

  <arg name="world_name" default="intersection"/>
  <!-- <arg name="count" default="4"/> -->

  <!-- create the world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find ackermann_vehicle_gazebo)/worlds/$(arg world_name).world"/>
    <arg name="gui" value="true"/>
    <arg name="debug" value="false"/>
    <arg name="paused" value="false"/>
    <arg name="headless" value="false"/>
    <arg name="use_sim_time" value="true"/>
  </include>

  <!-- spawn the vehicles -->
  <group ns="car1">
    <include
      file="$(find ackermann_vehicle_gazebo)/launch/ackermann_vehicle.launch">
      <arg name="ugv_id" value="car1"/>
      <arg name="x" default="-1.3"/>
      <arg name="y" default="6.2"/>
      <arg name="z" default="0.1"/>
      <arg name="roll" default="0.0"/>
      <arg name="pitch" default="0.0"/>
      <arg name="yaw" default="-1.6"/>
    </include>
  </group>

  <group ns="car2">
    <include
      file="$(find ackermann_vehicle_gazebo)/launch/ackermann_vehicle.launch">
      <arg name="ugv_id" value="car2"/>
      <arg name="x" default="-6.2"/>
      <arg name="y" default="1.3"/>
      <arg name="z" default="0.1"/>
      <arg name="roll" default="0.0"/>
      <arg name="pitch" default="0.0"/>
      <arg name="yaw" default="0.0"/>
    </include>
  </group>

  <group ns="car3">
    <include
      file="$(find ackermann_vehicle_gazebo)/launch/ackermann_vehicle.launch">
      <arg name="ugv_id" value="car3"/>
      <arg name="x" default="-1.3"/>
      <arg name="y" default="-6.2"/>
      <arg name="z" default="0.1"/>
      <arg name="roll" default="0.0"/>
      <arg name="pitch" default="0.0"/>
      <arg name="yaw" default="1.6"/>
    </include>
  </group>

  <group ns="car4">
    <include
      file="$(find ackermann_vehicle_gazebo)/launch/ackermann_vehicle.launch">
      <arg name="ugv_id" value="car4"/>
      <arg name="x" default="6.2"/>
      <arg name="y" default="1.3"/>
      <arg name="z" default="0.1"/>
      <arg name="roll" default="0.0"/>
      <arg name="pitch" default="0.0"/>
      <arg name="yaw" default="-3.15"/>
    </include>
  </group>

  <!-- launch other scripts -->

</launch>
